### PR TITLE
docs: docs for scheduled triggers (close #4913)

### LINKED ISSUE #4913: add docs for scheduled triggers
- cron triggers
- scheduled events

### PR DESCRIPTION
**Pull request by sezgi**
### Description
New page for `Scheduled triggers` under `Event Triggers`

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR. If no changelog is required, then add the `no-changelog-required` label.

### Affected components

- [x] Docs

### Related Issues
#4913

### CODE REVIEW

File: docs/graphql/manual/event-triggers/scheduled.rst [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+   :alt: Adding a cron trigger
+
+Step 2. Define the cron trigger
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the form opened by the above step, fill out these fields and click ``Create``:
+
+- **Name**: Create a name for the cron trigger.
+- **Webhook**: Enter the HTTP endpoint that should be triggered.
+- **Cron schedule**: Enter a schedule for the cron. You can use the link provided to help `build a cron expression <https://crontab.guru/#*_*_*_*_*>`__, or use the ``Frequently used crons`` dropdown as a shortcut. Cron events are created based on the UTC timezone.
```
Comment (marionschleifer) [inline docs/graphql/manual/event-triggers/scheduled.rst:41]: Can the link be on the text `link provided`? Also where is the link provided?

File: docs/graphql/manual/event-triggers/scheduled.rst [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+.. thumbnail:: /img/graphql/manual/event-triggers/define-cron.png
+   :alt: Defining a cron trigger
+   :width: 660px
+
+In this example, we're creating a cron trigger called ``eod_reports``, to trigger the webhook ``https://mywebhook.com/eod``. The cron schedule is set to ``0 22 * * 1-5``, which means "At 22:00 on every day-of-week from Monday through Friday" (you can check this `here <https://crontab.guru/#0_22_*_*_1-5>`__).
+
+Step 3. Define advanced options (Optional)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you like, you can also define these values by expanding the ``Advanced`` section:
```
Comment (marionschleifer) [inline docs/graphql/manual/event-triggers/scheduled.rst:53]:
```suggestion
If you like, you can also define the following values by expanding the ``Advanced`` section:
```

File: docs/graphql/manual/event-triggers/scheduled.rst [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+.. thumbnail:: /img/graphql/manual/event-triggers/pending-cron.png
+   :alt: The tabs
+
+One-off scheduled events
+------------------------
+
+One-off scheduled events are used to reliably trigger a HTTP webhook to run custom business logic at a particular timestamp. For example, you can create a scheduled event to send a reminder email two weeks after a user signs up.
+
+You can schedule an event from your backend using the :ref:`metadata API <create_scheduled_event>`, or through the Hasura console.
```
Comment (marionschleifer) [inline docs/graphql/manual/event-triggers/scheduled.rst:77]: This should ideally be done with tabs like in this PR: https://deploy-preview-4948--hasura-docs.netlify.app/graphql/manual/actions/create.html#step-1-define-your-mutation-and-associated-types. Also for the cron triggers above. But I guess this is not the most important thing at the moment. I can also add these to the tabs PR after this has been deployed. Just to keep in mind 🙂

File: docs/graphql/manual/event-triggers/scheduled.rst [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+- Click ``One-off Scheduled Events``.
+- Click ``Schedule an event``.
+
+.. thumbnail:: /img/graphql/manual/event-triggers/one-off.png
+   :alt: Adding a one-off scheduled event
+
+Step 2. Define the scheduled event
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the form opened by the above step, fill out these fields and click ``Create scheduled event``:
```
Comment (marionschleifer) [inline docs/graphql/manual/event-triggers/scheduled.rst:94]:
```suggestion
In the form opened by the above step, fill out the following fields:
```

File: docs/graphql/manual/event-triggers/scheduled.rst [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+If you like, you can also define these values by expanding the ``Advanced`` section:
+
+- **Headers**: List of headers to be sent to the webhook.
+- **Retry configuration**: In case the call to the webhook fails.
+- **Comment**: Custom description of the scheduled trigger.
+
+.. thumbnail:: /img/graphql/manual/event-triggers/advanced-one-off.png
+   :alt: Defining advanced options
+   :width: 809px
```
Comment (marionschleifer) [inline docs/graphql/manual/event-triggers/scheduled.rst:115]:
```suggestion
   :width: 809px
   
Click ``Create scheduled event``.
```

File: docs/graphql/manual/event-triggers/scheduled.rst (hunk 3 above)
Comment (sezgi) [reply docs/graphql/manual/event-triggers/scheduled.rst:77]: Where do you find the info for the other tabs? (This is the part of the docs process that I don't get yet -- where to find the info needed)

File: docs/graphql/manual/event-triggers/scheduled-triggers.rst [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+Schedule & logs
+^^^^^^^^^^^^^^^
+
+Once you've created your cron trigger, you can see ``Pending events``, ``Processed events``, and ``Invocation logs`` in their respective tabs.
+
+.. thumbnail:: /img/graphql/manual/event-triggers/pending-cron.png
+   :alt: The tabs
+
+One-off scheduled events
```
Comment (tirumaraiselvan) [inline docs/graphql/manual/event-triggers/scheduled-triggers.rst:75]: This should mention the metadata API `create_scheduled_event` as that is the primary way that one-off triggers would be used.

File: docs/graphql/manual/event-triggers/scheduled-triggers.rst [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+- **Headers**: List of headers to be sent to the webhook.
+- **Retry configuration**: In case the call to the webhook fails.
+- **Comment**: Custom description of the scheduled trigger.
+
+.. thumbnail:: /img/graphql/manual/event-triggers/advanced-one-off.png
+   :alt: Defining advanced options
+   :width: 809px
+
+Schedule & logs
```
Comment (tirumaraiselvan) [inline docs/graphql/manual/event-triggers/scheduled-triggers.rst:120]: Schedule & logs? Did you mean Scheduled events & logs?

File: docs/graphql/manual/event-triggers/scheduled.rst (hunk 3 above)
Comment (marionschleifer) [reply docs/graphql/manual/event-triggers/scheduled.rst]: @sezgi the API reference (API tab) for this already exists: https://hasura.io/docs/1.0/graphql/manual/api-reference/schema-metadata-api/scheduled-triggers.html.

And for the CLI, you can just run a Hasura instance locally and check what metadata gets added and then document it as shown in the CLI tab in this PR.

Let me know if you have more questions 🙂

File: docs/graphql/manual/event-triggers/scheduled-triggers.rst (hunk 7 above)
Comment (marionschleifer) [reply docs/graphql/manual/event-triggers/scheduled-triggers.rst:291]: It means the schedule as in when the events get executed: https://deploy-preview-5141--hasura-docs.netlify.app/graphql/manual/event-triggers/scheduled-triggers.html#schedule-logs.

**Comment by marionschleifer** [comment]
@sezgi I made the following changes to this PR:

- Added tabs for CLI / API where required
- In the screenshot for defining a cron trigger I added a `{}` to the payload field because there is an `invalid json` error if it's empty
- In the screenshot for defining a scheduled event, I added a date that is in the future. If users use the exact date that was provided, the events would go directly to "Processed events" because it's in the past
- Other small changes

cc @rikinsk

**Comment by rikinsk** [comment]
The scheduled triggers page being under event triggers feels a little odd and inaccurate to me. Event triggers (which should ideally be data triggers now) and Scheduled triggers should be at the same hierarchical level.

I would propose the following:
- Make `Scheduled triggers` as a top level directory
- Make `Cron triggers` and `One-off Scheduled Events` as sub-directories of `Scheduled triggers`. 
- Lets avoid having a single page for explaining all the concepts and instead try to follow the structure we currently have for Event triggers (even if we dont have all the relevant pages yet. We will have to add them soon anyway). This will avoid the situation we have with Remote schemas where we are now having to split the large single page into sub pages as the content has gotten too much.

Eventually we should also rename Event triggers to Data triggers for clarity I guess. But this is tricky for legacy reasons and needs to be thought out a bit more. Hence we can do this at a later point.

Other miscellaneous notes.
- Lets keep our page hierarchy, TOC and headings as clear as we can. e.g. In the below page, i dont really understand what the steps are for. An intermediate section called `Creating a cron trigger` needs to exist for this to make sense.
Screenshot from 2020-06-30 17-34-18

**Comment by sezgi** [comment]
@rikinsk please see update

**Comment by rikinsk** [comment]
@marionschleifer there has been some restructuring of pages done. I've taken a brief look but would be great if you take a look as well and give the final sign off

**Pull request merged**